src/admin.rs

```diff
@@ -406,61 +395,39 @@ fn change_log_level(reset: bool, level: &str) -> Response<Full<Bytes>> {
     }
 }
 
-#[cfg(feature = "gperftools")]
-async fn handle_gprof(_req: Request<Incoming>) -> Response<Full<Bytes>> {
-    const FILE_PATH: &str = "/tmp/profile.prof";
-    PROFILER.lock().unwrap().start(FILE_PATH).unwrap();
-
-    tokio::time::sleep(Duration::from_secs(10)).await;
-    PROFILER.lock().unwrap().stop().unwrap();
-
-    let mut buffer = Vec::new();
-    File::open(FILE_PATH)
-        .await
-        .unwrap()
-        .read_to_end(&mut buffer)
-        .await
-        .unwrap();
-    Response::builder()
-        .status(hyper::StatusCode::OK)
-        .body(buffer.into())
-        .unwrap()
-}
-
-#[cfg(not(feature = "gperftools"))]
-async fn handle_gprof(_req: Request<Incoming>) -> Response<Full<Bytes>> {
-    Response::builder()
-        .status(hyper::StatusCode::INTERNAL_SERVER_ERROR)
-        .body("gperftools not enabled".into())
-        .unwrap()
-}
-
-#[cfg(feature = "gperftools")]
-async fn handle_gprof_heap(_req: Request<Incoming>) -> Response<Full<Bytes>> {
-    const FILE_PATH: &str = "/tmp/profile.prof";
-    HEAP_PROFILER.lock().unwrap().start(FILE_PATH).unwrap();
-
-    tokio::time::sleep(Duration::from_secs(10)).await;
-    HEAP_PROFILER.lock().unwrap().stop().unwrap();
-
-    let mut buffer = Vec::new();
-    File::open(FILE_PATH)
-        .await
-        .unwrap()
-        .read_to_end(&mut buffer)
-        .await
-        .unwrap();
-    Response::builder()
-        .status(hyper::StatusCode::OK)
-        .body(buffer.into())
-        .unwrap()
+#[cfg(feature = "jemalloc")]
+async fn handle_jemalloc_pprof_heapgen(_req: Request<Incoming>) -> Response<Full<Bytes>> {
+    const FILE_PATH: &str = "/tmp/jemalloc-pprof-heap.prof";
+    // make sure the file exists/is writable before handing to gprof
+    File::create(FILE_PATH).await.unwrap();
```

Comment:
can we stream it back instead of file? is the format the "pprof" format that can be read by `go tool pprof` or some other format?

Reply:
`go tool pprof` - starting a PROFILING.md